tests/bls/test_hash_to_curve.py

```diff
@@ -86,3 +86,27 @@ def test_hash_to_G2(msg, x, y, H):
 
     assert x == result_x
     assert y == result_y
+
+
+@pytest.mark.parametrize('value, expected', [
+    (0x00, 0),
+    (0x10, 0),
+    (0x01, 1),
+])
+def test_FQ_sgn0(value, expected):
+    x = FQ(value)
+    assert x.sgn0 == expected
+    if x.n != 0:
+        assert x.sgn0 != (-x).sgn0
+
+
+@pytest.mark.parametrize('value, expected', [
+    ([0x00, 0x00], 0),
+    ([0x10, 0x00], 0),
+    ([0x01, 0x00], 1),
+    ([0x01, 0x01], 1),
+    ([0x10, 0x10], 0),
+])
+def test_FQ2_sgn0(value, expected):
```

Comment:
This is a good idea. Where did you get the test vectors from? I think there should also be another test with the same parameters, but for the general case (`FQP`)